code/modules/clothing/chameleon.dm

```diff
@@ -580,6 +580,40 @@
 	var/datum/action/item_action/chameleon/drone/randomise/randomise_action = new(src)
 	randomise_action.UpdateButtonIcon()
 
+/datum/action/item_action/chameleon/tongue_change
+	name = "Tongue Change"
+	icon_icon = 'icons/obj/surgery.dmi'
+	button_icon_state = "tonguebone"
+	var/list/tongue_list = list()
+
+/datum/action/item_action/chameleon/tongue_change/proc/generate_tongue_list()
+	var/obj/item/I
+	var/tongue_name
+	var/static/list/predefined_tongues = typesof(/obj/item/organ/tongue)
+	for(var/V in predefined_tongues)
+		I = V
+		if((initial(I.item_flags) & ABSTRACT) || !initial(I.icon_state))
+			continue
+		tongue_name = "[initial(I.name)] ([initial(I.icon_state)])"
+		tongue_list[tongue_name] = I
```

Comment:
Use better variable names. human/H is acceptable but stuff like var/V isn't. It needs to be indicative of what it is, is it a path, it is an object? Something like `tongue_path` would make sense